client/order/cancel-authorization-confirm-modal/index.tsx

```diff
@@ -0,0 +1,134 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import * as React from 'react';
+import { __ } from '@wordpress/i18n';
+import { Button } from '@wordpress/components';
+import interpolateComponents from '@automattic/interpolate-components';
+import { useState } from '@wordpress/element';
+/**
+ * Internal dependencies
+ */
+import ConfirmationModal from 'wcpay/components/confirmation-modal';
+
+interface CancelAuthorizationConfirmationModalProps {
+	originalOrderStatus: string;
+}
+
+const CancelAuthorizationConfirmationModal: React.FunctionComponent< CancelAuthorizationConfirmationModalProps > = ( {
+	originalOrderStatus,
+} ) => {
+	const [
+		isCancelAuthorizationConfirmationModalOpen,
+		setIsCancelAuthorizationConfirmationModalOpen,
+	] = useState( true );
+
+	const closeModal = (): void => {
+		setIsCancelAuthorizationConfirmationModalOpen( false );
+	};
+
+	const handleCancelOrder = (): void => {
+		const orderEditForm: HTMLFormElement | null = document.querySelector(
```

Comment:
A bit more cryptic, but clearer connection to the status field: `document.querySelector( '#order_status' )?.closest( 'form' ) || null;`.

My thinking is if the select ID changes, it'll be easier to find usage. Also, we reinforce our logic towards changes in the form ID/structure.

Reply:
Addressed in e8ff054c6